### ISSUE TITLE
CA1507 should allow to configure symbols to exclude

### ISSUE THREAD

**Issue by volleyms** [dotnet/roslyn-analyzers#6886]
### Analyzer

**Diagnostic ID**: [CA1507](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1507)

### Describe the improvement

Allow to configure symbols which then are excluded from CA1507, e.g. `Newtonsoft.Json.JsonPropertyAttribute` or `System.Xml.Serialization.XmlElement`, since it is valid (if not even preferred) to keep externally serialized names stable and robust against refactorings (renaming). The following example is detected as CA1507 unconditionally, unfortunately:
```
[JsonProperty("Id")] // CA1507
public Guid Id { get; set; }
```

### Describe suggestions on how to achieve the rule

Just as possible for other diagnostics, allow to configure symbols to exclude (maybe, to exclude methods is sufficient). Here the exclusion shall then ignore parameters when called.
```
dotnet_diagnostic.CA1507.excluded_symbol_names = M:Newtonsoft.Json.JsonPropertyAttribute.#ctor
```
